### PR TITLE
feat(iterm2): add shell integration script

### PR DESCRIPTION
**Pull request by reegnz**
Mirror the official script and source it with the iterm2 plugin.

The integration script brings features like automatic marks on each prompt, drag and drop scp support, or automatic profile switching.

## Standards checklist:

- [x] The PR title is descriptive.
- [x] The PR doesn't replicate another PR which is already open.
- [x] I have read the contribution guide and followed all the instructions.
- [x] The code follows the code style guide detailed in the wiki.
- [x] The code is mine or it's from somewhere with an MIT-compatible license.
- [x] The code is efficient, to the best of my ability, and does not waste computer resources.
- [x] The code is stable and I have tested it myself, to the best of my abilities.
- [x] If the code introduces new aliases, I provide a valid use case for all plugin users down below.

## Changes:

- adds script from https://iterm2.com/documentation-shell-integration.html
- sources that script

### CODE REVIEW

**Comment by reegnz** [comment]
@avivrosenberg

**Comment by carlosala** [comment]
in terms of license, it is ok. Even though, I'd make it opt-in, using `zstyle`. You can see how to do this in `nvm` plugin, for example.

**Comment by reegnz** [comment]
I'm not sure about the opt-in, I think the shell-integration is quite essential when using iterm2. If the concern is that existing users that use iterm2 plugin don't want to be surprised by having it enabled, that's fair. I don't think it's backward-incompatible change though, so it won't break anything IMHO.

So sure, I can add a zstyle if it makes it easier to get merged.

File: plugins/iterm2/README.md [hunk 1]
```diff
@@ -2,11 +2,20 @@
...
 ```
 plugins=(... iterm2)
 ```
 
+Optionally, the plugin also applies the [Shell Integration Script for iTerm2](https://iterm2.com/documentation-shell-integration.html).
+You can enable the integration with zstyle. It's important to add this line 
+before the line sourcing oh-my-zsh:
+
+```
+zstyle :omz:plugins:iterm2 shell-integration true
```
Comment (carlosala) [inline plugins/iterm2/README.md:16]: Please change `true` with `yes`.

**Review by carlosala — CHANGES_REQUESTED** [review]

File: plugins/iterm2/README.md (hunk 1 above)
Comment (reegnz) [reply plugins/iterm2/README.md]: Fixed

**Comment by carlosala** [comment]
Thanks for your contribution!

**Pull request merged**